openlibrary/core/cache.py

```diff
@@ -177,20 +177,20 @@ def join_threads(self):
             thread.join()
 
     def encode_args(self, args, kw=None):
+        """Encodes arguments to construct the memcache key."""
         kw = kw or {}
-        """Encodes arguments to construct the memcache key.
-        """
+
         # strip [ and ] from key
         a = self.json_encode(list(args))[1:-1]
 
         if kw:
-            return a + "-" + self.json_encode(kw)
+            return f"{hash(a)}${hash(self.json_encode(kw))}"
```

Comment:
I think we'll want to use an md5 hash function here, apparently python's `hash` is non-deterministic between sessions.